### ISSUE TITLE
`useLogger` returns a new object on every render

### ISSUE THREAD

**Issue by Luluno01** [axiomhq/next-axiom#164]
## Cause

The client-side hook `useLogger` returns a new, unmemorized logger object on every render.

https://github.com/axiomhq/next-axiom/blob/b85adae38d081fa536d6d0a1289d5062e7e7472b/src/hooks.ts#L20-L21

## Details

It is legit and common for users to use a logger inside `useEffect` or `useCallback` hooks. In order to make full use of eslint's static hook dependency analysis to rule out any potentially unwanted bug due to missing hook dependency, it requires the programmer to explicitly add all dependencies used by the hook, including the logger. The `useLogger` hook, however, returns a new logger object on every render, forcing users to manually memorize the logger or dangerously disable ESLint static hook dependency analysis with `// eslint-disable react-hooks/exhaustive-deps`. Otherwise, it either leads to infinite re-render or a linting error.

## Example

```TSX
'use client'
import { useEffect, useState } from 'react'
import { useLogger } from 'next-axiom'

export default function Blog({ id }: { id: string }) {
  const log = useLogger()
  const [ blog, setBlog ] = useState<{ title: string, body: string }>()
  useEffect(() => {
    async function fetchBlog() {
      const res = await fetch(`/api/blog/${id}`)
      setBlog(await res.json()) // Triggers re-render
    }
    fetchBlog().catch(err => log.error('Failed to fetch blog', { err }))
  }, [ id, log ]) // `log` changes after `setBlog` triggering a re-render => infinite loop
  return <>
    <h1>{blog.title}</h1>
    <p>{blog.body}</p>
  <>
}
```

**Comment by schehata** [comment]
good catch! thanks for sharing this with us, will probably use a ref and test if that works out. will get back to you when PR is ready, thanks again.

**Comment by c-ehrlich** [comment]
Thanks for reporting. This should be fixed by #169 which is included in v1.1.1

Note that you'll still get two loggers in strict mode as the component fully unmounts. Closing for now but feel free to reopen if there are any issues with the fix :)
